vsgenstubs4/init.py

```diff
@@ -0,0 +1,579 @@
+    name: str
+    description: str
+    bound: BoundSignature
+
+    @classmethod
+    def from_namespace(cls, namespace: str, cores: Sequence[CoreLike]) -> 'PluginMeta':
+        try:
+            plugin = cores[0].__getattr__(namespace)
+        except BaseException:
+            raise ValueError(f'Invalid namespace! Plugin not found: "{namespace}"')
+
+        return PluginMeta(
+            plugin.namespace, plugin.name, BoundSignature(plugin.namespace, cores)
+        )
+
+    def _str_(self, kind: str, __x: tuple[object, ...]) -> bool:
+        return getattr(str, kind)(self.name.lower(), str(__x[0]).lower())
+
+    def __gt__(self, x: 'PluginMeta', /) -> bool: return self._str_('__gt__', x)  # type: ignore[override]
+    def __lt__(self, x: 'PluginMeta', /) -> bool: return self._str_('__lt__', x)  # type: ignore[override]
+    def __ge__(self, x: 'PluginMeta', /) -> bool: return self._str_('__ge__', x)  # type: ignore[override]
+    def __le__(self, x: 'PluginMeta', /) -> bool: return self._str_('__le__', x)  # type: ignore[override]
+
+
+def retrieve_plugins(
+    args: Namespace, core: vs.Core, cores: Iterable[CoreLike]
+) -> Iterator[PluginMeta]:
+    lower_plugins = list(map(str.lower, args.plugins)) if args.plugins else []
+
+    if lower_plugins:
+        find_plugins = lower_plugins.copy()
+
+        for p in core.plugins():
+            ns_lower = p.namespace.lower()
+
+            if ns_lower in find_plugins:
+                find_plugins.remove(ns_lower)
+
+        if find_plugins:
+            missing_plugins = ', '.join(find_plugins)
+
+            raise ModuleNotFoundError(
+                'Can\'t generate stubs for specific plugins, these are missing in your installation:'
+                f'\n\t"{missing_plugins}"'
+            )
+
+    for p in core.plugins():
+        if lower_plugins and (p.namespace.lower() not in lower_plugins):
+            continue
+
+        yield PluginMeta(p.namespace, p.name, BoundSignature(p.namespace, cores))
+
+
+implementation_start = '# implementation'
+implementation_end = '# end implementation'
+
+
+class Implementation(NamedTuple):
+    plugin: PluginMeta
+    content: List[str]
+
+    @classmethod
+    def from_namespace(cls, namespace: str, cores: Sequence[CoreLike]) -> 'Implementation':
+        return Implementation(PluginMeta.from_namespace(namespace, cores), [])
+
+    @staticmethod
+    def get_name(plugin: PluginMeta, core_name: str, /) -> str:
+        return f'_Plugin_{plugin.name}_{core_name}_Bound'
+
+    def __gt__(self, x: 'Implementation', /) -> bool: return self.plugin.__gt__(x.plugin)  # type: ignore[override]
+    def __lt__(self, x: 'Implementation', /) -> bool: return self.plugin.__lt__(x.plugin)  # type: ignore[override]
+    def __ge__(self, x: 'Implementation', /) -> bool: return self.plugin.__ge__(x.plugin)  # type: ignore[override]
+    def __le__(self, x: 'Implementation', /) -> bool: return self.plugin.__le__(x.plugin)  # type: ignore[override]
```

Comment:
[Why not use functools.total_odering?](https://docs.python.org/3/library/functools.html#functools.total_ordering)

Reply:
It's slower, and I don't see the need.
Would've done something with inheritance but i'm working with NamedTuples (again, for speed).